tests/publiccloud/ipa.pm

```diff
@@ -41,21 +42,44 @@ sub create_secgroup {
     return find_secgroup($name);
 }
 
+sub create_ssh_key {
+    my ($prefix, $out_file) = @_;
+
+    for my $i (0 .. 9) {
+        my $key_name = $prefix . "_" . $i;
+        my $cmd      = "aws ec2 create-key-pair --key-name '" . $key_name . "' --query 'KeyMaterial' --output text > " . $out_file;
+        if (script_run($cmd) == 0) {
+            return $key_name;
+        }
+    }
+    die("Unable to create SSH key on aws with prefix '$prefix'");
+}
+
 sub save_logs {
+    return if (script_run("test -d ipa_results") != 0);
+
     my $output = script_output("find ipa_results -type f");
     for my $file (split(/\n/, $output)) {
         if ($file =~ m'ipa_results/ec2/ami-[0-9a-z]+/i-[0-9a-z]+/[0-9]{14}\.(log|results)') {
             upload_logs($file, failok => 1);
+            if ($file =~ /results$/) {
+                parse_extra_log(IPA => $file);
+            }
         }
     }
 }
 
 sub run {
+    my ($self) = @_;
 
     select_virtio_console();
 
     die "Public cloud provider isn't supported" unless check_var('PUBLIC_CLOUD_PROVIDER', "EC2");
 
+    if (check_var('DISTRI', 'sle')) {
+        my $modver = get_required_var("VERSION") =~ s/-SP\d+//gr;
```

Comment:
Please use single-ticks by default unless you want to evaluate variables within strings. I wonder why we need this special treatment of the version here. If this is something that is necessary in more places we can also move it into the `add_suseconnect_product` call

Reply:
The module version of public-cloud doesn't contain the service-pack number. I don't see a reason why some modules have a SP number and others not.

Reply:
Thanks for pointing to that link.